src/System.Threading.Thread/tests/ThreadTests.cs

```diff
@@ -427,6 +429,89 @@ public static void CurrentPrincipalTest()
             });
         }
 
+        [Fact]
+        public static void CurrentPrincipalContextFlowTest()
+        {
+            ThreadTestHelpers.RunTestInBackgroundThread(async () =>
+            {
+                Thread.CurrentPrincipal = new ClaimsPrincipal();
+
+                await Task.Run(async() => {
+
+                    Assert.IsType<ClaimsPrincipal>(Thread.CurrentPrincipal);
+
+                    await Task.Run(async() => 
+                    {
+                        Assert.IsType<ClaimsPrincipal>(Thread.CurrentPrincipal);
+
+                        Thread.CurrentPrincipal = new GenericPrincipal(new GenericIdentity("name"), new string[0]);
+
+                        await Task.Run(() =>
+                        {
+                            Assert.IsType<GenericPrincipal>(Thread.CurrentPrincipal);
+                        });
+
+                        Assert.IsType<GenericPrincipal>(Thread.CurrentPrincipal);
+                    });
+
+                    Assert.IsType<ClaimsPrincipal>(Thread.CurrentPrincipal);
+                });
+
+                Assert.IsType<ClaimsPrincipal>(Thread.CurrentPrincipal);
+            });
+        }
+
+        [Fact]
+        public static void CurrentPrincipalContextFlowTest_NotFlow()
+        {
+            ThreadTestHelpers.RunTestInBackgroundThread(async () =>
+            {
+                Thread.CurrentPrincipal = new ClaimsPrincipal();
+
+                Task task;
+                using(ExecutionContext.SuppressFlow())
+                {
+                    Assert.True(ExecutionContext.IsFlowSuppressed());
+
+                    task = Task.Run(() => 
+                    {
+                        // Default PrincipalPolicy for netcoreapp is null and for netfx is ClaimsPrincipal
+                        Assert.True(Thread.CurrentPrincipal is null || Thread.CurrentPrincipal is ClaimsPrincipal);
```

Comment:
This is going to pass on netcoreapp even if we have the wrong behavior of the principal flowing.

I suggest just making the test SkipOnTargetFramework, skipping it for netfx, and then making this logic as tight as we can to correctly validate the behavior we expect on netcoreapp.

Reply:
It was a request https://github.com/dotnet/corefx/pull/34747#discussion_r250276887, to be sure that the behaviour is the same. Maybe I could fix asserts with https://github.com/dotnet/corefx/blob/master/src/CoreFx.Private.TestUtilities/src/System/PlatformDetection.cs#L24

Reply:
> It was a request #34747 (comment), to be sure that the behaviour is the same

Yes, but it's no longer validating that the behavior is the same.

Reply:
yes I agree test is wrong right now, I'll fix with `PlatformDetection` helper.

Reply:
fixed